client/src/features/projectsV2/fields/ProjectVisibilityFormField.tsx

```diff
@@ -39,16 +46,62 @@ export default function ProjectVisibilityFormField<T extends FieldValues>({
         control={control}
         name={name}
         render={({ field }) => (
-          <Input
-            className={cx("form-control", errors.visibility && "is-invalid")}
-            data-cy="project-visibility"
-            id="project-visibility"
-            type="select"
-            {...field}
-          >
-            <option value="public">Public</option>
-            <option value="private">Private</option>
-          </Input>
+          <div className="d-flex flex-row gap-5 my-3">
+            <div className="d-flex align-items-center gap-2">
+              <Input
+                type="radio"
+                className={cx(
+                  "form-control",
+                  errors.visibility && "is-invalid",
+                  "p-0",
+                  "mt-0"
+                )}
+                data-cy="project-visibility"
+                id="project-visibility-public"
+                {...field}
+                value="public"
+                checked={field.value === "public"}
+              />
+              <label
+                onClick={() => setVisibility("public")}
```

Comment:
Do not use a `onClick()` action for this, use the `for` attribute. This makes clicking on the label work as intended. Note that `onClick` also creates a bug where the "Update" button does not become active when changing the visibility.

```suggestion
                for="project-visibility-public"
```